packages/firebase_dynamic_links/firebase_dynamic_links_platform_interface/lib/src/platform_interface/platform_interface_firebase_dynamic_links.dart

```diff
@@ -0,0 +1,105 @@
+// Copyright 2021, the Chromium project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:firebase_core/firebase_core.dart';
+import 'package:meta/meta.dart';
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import 'package:firebase_dynamic_links_platform_interface/firebase_dynamic_links_platform_interface.dart';
+
+import '../method_channel/method_channel_firebase_dynamic_links.dart';
+
+/// Defines an interface to work with Dynamic Links across platforms
+abstract class FirebaseDynamicLinksPlatform extends PlatformInterface {
+  /// The [FirebaseApp] this instance was initialized with.
+  @protected
+  final FirebaseApp? appInstance;
+
+  /// Create an instance using [app]
+  FirebaseDynamicLinksPlatform({this.appInstance}) : super(token: _token);
+
+  /// Returns the [FirebaseApp] for the current instance.
+  FirebaseApp get app {
+    return appInstance ?? Firebase.app();
+  }
+
+  static final Object _token = Object();
+
+  /// Create an instance using [app] using the existing implementation
+  factory FirebaseDynamicLinksPlatform.instanceFor({required FirebaseApp app}) {
+    return FirebaseDynamicLinksPlatform.instance.delegateFor(app: app);
+  }
+
+  /// The current default [FirebaseDynamicLinksPlatform] instance.
+  ///
+  /// It will always default to [MethodChannelFirebaseDynamicLinks]
+  /// if no other implementation was provided.
+  static FirebaseDynamicLinksPlatform get instance {
+    return _instance ??= MethodChannelFirebaseDynamicLinks(app: Firebase.app());
```

Comment:
I don't think this should be passing app over here: `MethodChannelFirebaseDynamicLinks(app: Firebase.app())` -> `MethodChannelFirebaseDynamicLinks()`.

The instance getter creates a single instance of `MethodChannelFirebaseDynamicLinks`, and when `instanceFor` is called, it calls the single instance, and then chains `delegateFor` which provdies the app (`return FirebaseDynamicLinksPlatform.instance.delegateFor(app: app);`).